pkg/agent/client.go

```diff
@@ -421,7 +429,7 @@ func (a *Client) Serve() {
 
 			ctx, ok := a.connManager.Get(data.ConnectID)
 			if ok {
-				ctx.dataCh <- data.Data
+				ctx.send(data.Data)
```

Comment:
Walter and I talked about moving this dataCh write into the `connManager.Get`, and move the `close(dataCh)` at line 404 into the `connManager.Delete(connID)`. Then we use the `connManager.mu` to avoid a race. The problem with this solution is that holding `connManager.mu` affects all connections, not just this connection.

The other idea we had was adding a new lock to protect the dataCh. And since that Serve() function is single-threaded, we will not have any read contention. The only read-write contention will be when the cleanup function (which closes the dataCh) is called by another goroutine. The performance cost of adding such a low contention lock should be moderate, but we don't have benchmarks to validate this, so let's stick with this panic recovery fix for now.  @cheftako can you add a TODO about using a lock? Thanks.